### PR TITLE
merge-images accepts Buffer

### PR DESCRIPTION
**Pull request by k-yle**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
      &nbsp;
      It's currently an array of strings, but the [readme](https://github.com/lukechilds/merge-images#images) says that it's an:
     >  "Array of [valid image sources](https://github.com/Automattic/node-canvas#imagesrc) for `new Image()`."     
     
- [x] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~

### CODE REVIEW

File: types/merge-images/index.d.ts [hunk 1]
```diff
@@ -14,7 +17,9 @@ declare function mergeImages(
 ): Promise<string>;
 
 declare namespace mergeImages {
-    type ImageSource = string | { src: string; x?: number; y?: number; opacity?: number };
+    type Image = string | Buffer;
```
Comment (BendingBender) [inline types/merge-images/index.d.ts:20]: I don't get it. You can set a `Buffer` as a value of `img.src`? What `Image` implementation is it? Does it work in a browser?

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: types/merge-images/index.d.ts (hunk 1 above)
Comment (k-yle) [reply types/merge-images/index.d.ts:20]: > You can set a `Buffer` as a value of `img.src`. 


The docs say that `img.src` can be "Array of valid image sources for new Image()."

> "Does it work in a browser?"

No, `Buffer` is a nodejs type

Comment (k-yle) [reply types/merge-images/index.d.ts:20]: > "What `Image` implementation is it?"

@BendingBender this is for nodejs, where you have to use the `Image` implementation from [node-canvas](https://npm.im/canvas). 

node-canvas's `Image` accepts `string | Buffer` - [see here](https://github.com/Automattic/node-canvas/blob/master/types/index.d.ts#L232-L236)

Comment (BendingBender) [reply types/merge-images/index.d.ts:20]: I think it's OK then.

**Review by BendingBender — APPROVED** [review]

**Comment by k-yle** [comment]
Ready to merge

**Pull request merged**